test/ctia/stores/es/crud_test.clj

```diff
@@ -192,3 +199,238 @@
+                                :aggregate-on "the-date"
+                                :granularity :week
+                                :timezone "+02:00"})))
+  (is (= {:metric
+          {:terms
+           {:field "disposition"
+            :size 20
+            :order {:_count :desc}}}}
+         (sut/make-aggregation {:agg-type :topn
+                                :aggregate-on "disposition"
+                                :limit 20
+                                :sort_order :desc})))
+  (is (= {:metric
+          {:cardinality
+           {:field "observable.value"
+            :precision_threshold 10000}}}
+         (sut/make-aggregation {:agg-type :cardinality
+                                :aggregate-on "observable.value"}))))
+
+(defn generete-sightings
+  [nb confidence title timestamp]
+  (repeatedly nb
+              #(assoc base-sighting
+                      :id (gen-random-uuid)
+                      :confidence confidence
+                      :created timestamp
+                      :title title)))
+
+(def timestamp-1 "2020-04-01T00:00:00.000Z")
+(def timestamp-2 "2020-05-01T00:00:00.000Z")
+(def title1 "this is title1 sighting")
+(def title2 "this is title2 sighting")
+
+(def high-t1-title1 (generete-sightings 60
+                                        "High"
+                                        title1
+                                        timestamp-1))
+(def high-t2-title2 (generete-sightings 20
+                                        "High"
+                                        title2
+                                        timestamp-2))
+(def medium-t1-title1 (generete-sightings 10
+                                          "Medium"
+                                          title1
+                                          timestamp-1))
+(def low-t2-title2 (generete-sightings 5
+                                       "Low"
+                                       title2
+                                       timestamp-2))
+
+(def search-metrics-entities
+  (concat high-t1-title1
+          high-t2-title2
+          medium-t1-title1
+          low-t2-title2))
+
+(deftest handle-query-string-search-test
+  (testing "handle search shall properly apply query and params"
+    (let [es-conn-state (-> (init/init-es-conn! props-not-aliased)
+                            (update :props assoc :default_operator "AND"))
+          _ (create-fn es-conn-state
+                       search-metrics-entities
+                       ident
+                       {:refresh "true"})
+          query-string "title1"
+          date-range {:created {:gte timestamp-1
+                                :lt timestamp-2}}
+          filter-map {:confidence "High"}
+          search (fn [q params]
+                   (search-fn es-conn-state q ident params))]
+
+      (testing "Properly handle different search query options"
+        (is (= (count (concat high-t1-title1
```

Comment:
Say that for whatever reason (refactorings...), both hands of the `=` eval'ed to 0.

Then we'd be comparing `(= 0 0)` which would pass but lose its original intent.

Accordingly I'd place `assert`s here and there.